### DISCUSSION TITLE
A more detailed Signature property for Ops

### DISCUSSION THREAD

**Discussion by ferrine** [pymc-devs/pytensor#89] [Ideas]
https://github.com/pymc-devs/pytensor/blob/2ebfbf1c0d26b81b12304e154e51cd5d9ea91248/pytensor/tensor/random/basic.py#L398

Will it be useful to make something like this as an Op property? I'm thinking of vectorize rewrites and what they should rely on

**Comment by michaelosthege** [comment]
If broadcasting needs to be accountable for, this is exactly what `infer_shape` does, right?

**Reply by ricardoV94** [reply]
I am not sure I understand the question. This just says how an Op should be vectorized, it doesn't tell what the core shape will be.

This together with infer_shape of the core Op would tell you what the shape of such vectorized Op would be with batched inputs, without having to modify that infer_shape logic.
